File: src/main/java/io/vertigo/shiny/components/media/youtube/ShinyYoutube.java
package io.vertigo.shiny.components.media.youtube;

import io.vertigo.core.lang.Assertion;
import io.vertigo.shiny.components.ShinyComponent;

public record ShinyYoutube(
		String title,
		String videoId) implements ShinyComponent {

	@Override
	public String type() {
		return "ShinyYoutube";
	}


	public ShinyYoutube {
		Assertion.check()
				.isNotBlank(videoId, "Video ID cannot be blank");
	}
}

File: banshee/src/models/ShinyYoutube.ts
import { ShinyComponent } from '../ShinyComponent';

export class ShinyYoutube implements ShinyComponent {
  title?: string;
  videoId: string;
  type: string = 'youtube';

  constructor(title: string, videoId: string) {
    this.title = title;
    this.videoId = videoId;
  }
}
